libs/service-portal/documents/src/components/DocumentCard/DocumentCard.tsx

```diff
@@ -1,29 +1,95 @@
-import React, { FC } from 'react'
-import { ActionCard, ActionMenuItem } from '@island.is/service-portal/core'
+import React, { FC, useEffect } from 'react'
+import { ActionCard } from '@island.is/service-portal/core'
 import { Document } from '@island.is/api/schema'
+import { useLazyDocumentDetail } from '@island.is/service-portal/graphql'
+import { useLocale } from '@island.is/localization'
+import { toast } from '@island.is/island-ui/core'
+
+const downloadAsPdf = (base64Pdf: string, fileName: string) => {
+  if (typeof window === 'undefined') {
+    return
+  }
+  const fakeLink = window.document.createElement('a')
+  const url = `data:application/pdf;base64,${base64Pdf}`
+  fakeLink.href = url
+  fakeLink.download = fileName
+  fakeLink.title = fileName
+  const clickHandler = () => {
+    setTimeout(() => {
+      URL.revokeObjectURL(url)
+      fakeLink.removeEventListener('click', clickHandler)
+    }, 150)
+  }
+  fakeLink.addEventListener('click', clickHandler, false)
+  fakeLink.click()
+}
+
+const openExternalDocument = (url: string) => {
+  if (typeof window === 'undefined') {
+    return
+  }
+  window.open(url, '_blank')
+}
 
 interface Props {
   document: Document
 }
 
 const DocumentCard: FC<Props> = ({ document }) => {
-  // TODO: This call is really heavy on the API
-  // const { data } = useDocumentDetail(document.id)
+  const fileName = `${document.subject}.pdf`
+  const { formatMessage } = useLocale()
+  const { fetchDocument, loading, data, error } = useLazyDocumentDetail(
+    document.id,
+  )
+
+  useEffect(() => {
+    if (data) {
+      handleOnFetch()
+    }
+  }, [data])
+
+  const handleOnFetch = () => {
+    if (data?.fileType === 'pdf' && data?.content) {
+      downloadAsPdf(data.content, fileName)
+      return
+    }
+    if (data?.url) {
+      openExternalDocument(data.url)
+      return
+    }
+    if (error && !loading) {
+      toast.error(
+        formatMessage({
+          id: 'sp.documents:documentCard-error-singleDocument',
+          defaultMessage: 'Ekki tókst að sækja skjal',
+        }),
+      )
+    }
+  }
+
+  const handleOnClick = () => {
+    if (loading) {
+      return
+    }
+    if (!data) {
+      fetchDocument()
+    }
+    handleOnFetch()
```

Comment:
Should this not be wrapped in an else statement?
Otherwise it seems that handleOnFetch would run twice if the data object is empty
Once when we call it here and once in the effect hook when data gets populated

Reply:
Good point, I will fix this.